src/lib.rs

```diff
@@ -771,6 +801,7 @@ where
     /// Open a shard file that stores `T` items.
     fn open<P: AsRef<Path>>(path: P) -> Result<ShardReaderSingle<T, S>, Error> {
         let f = File::open(path).unwrap();
+        advise_file_random(&f);
```

Comment:
Is this really random?  It looks like it reads serially, just from a block in the middle of the file.  Not sure that `FADV_RANDOM` will help you at all with this, and it would probably be better to just `FADV_WILLNEED` the block you actually need once you know what that is.

Reply:
Say we wrote N items with a buffer size of K when we created the file.  There will N/K sorted 'chunks' in the file, and we'll be simultaneously reading through those chunks. For a typical job N/K ~ 20 - 50.  So there's effectively N/K cursors in use simultaneously.  So any readahead the kernel does should be useful in the near future. But I'm not sure what affect all the jumping around will have.

Reply:
This seems more like a response to my other comment.  But for that you're declaring that your reads will be sequential within a particular region of the file, not for the entire file.  I _think_ that if you make separate `FADV_SEQUENTIAL` calls, the kernel will do prefetch as needed on a per-block basis.  If not, though, `FADV_NOREUSE` should take care of stuff for you.